src/mumble/Settings.cpp

```diff
@@ -928,7 +928,7 @@ void Settings::legacyLoad(const QString &path) {
 	LOADENUM(ceChannelDrag, "ui/drag");
 	LOADENUM(ceUserDrag, "ui/userdrag");
 	LOADENUM(aotbAlwaysOnTop, "ui/alwaysontop");
-	LOAD(bAskOnQuit, "ui/askonquit");
+	LOADENUM(wcqbWhenConnectedQuitBehaviour, "ui/whenconnectedquitbehaviour");
```

Comment:
In the legacy load, we don't make any changes. Its for legacy only and thus any settings processed by this by definition only know about the old askOnQuit setting.

Reply:
I am not sure yet. If I leave it like this (with bAskOnQuit), it won't compile obviously since the field does not exist anymore. And I get serialization issues with the new enum.

Reply:
Yes, you'll have to load the setting into a temporary boolean and then convert that into the approprate enum representation manually.